packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -30,6 +30,55 @@ class SizeAnalyzer {
 
   static const int tableWidth = 80;
 
+  /// Analyze the [aotSnapshot] in an uncompressed output directory.
+  Future<Map<String, dynamic>> analyzeAotSnapshot({
+    @required Directory outputDirectory,
+    @required File aotSnapshot,
+    @required bool silent,
+    String excludePath,
+  }) async {
+    if (!silent) {
+      logger.printStatus('▒' * tableWidth);
+      logger.printStatus('━' * tableWidth);
+    }
+    final _SymbolNode aotAnalysisJson = _parseDirectory(
+      outputDirectory,
+      outputDirectory.parent.path,
+      excludePath,
+    );
+
+    // Convert an AOT snapshot file into a map.
+    final Map<String, dynamic> processedAotSnapshotJson = treemapFromJson(
+      json.decode(aotSnapshot.readAsStringSync()),
+    );
+    final _SymbolNode aotSnapshotJsonRoot = _parseAotSnapshot(processedAotSnapshotJson);
+
+    if (!silent) {
+      for (final _SymbolNode firstLevelPath in aotAnalysisJson.children) {
+        _printEntitySize(
+          firstLevelPath.name,
+          byteSize: firstLevelPath.byteSize,
+          level: 1,
+        );
+        // Print the expansion of lib directory to show more info for `appFilename`.
+        if (firstLevelPath.name == fileSystem.path.basename(outputDirectory.path)) {
+          _printLibChildrenPaths(firstLevelPath, '', aotSnapshotJsonRoot);
+        }
+      }
+    }
+
+    if (!silent) {
+      logger.printStatus('▒' * tableWidth);
+    }
+
+    final Map<String, dynamic> apkAnalysisJson = aotAnalysisJson.toJson();
+
+    apkAnalysisJson['type'] = 'snapshot';
```

Comment:
Not sure what I should be writing here, and what paths are required. the APK code size analysis seems to hardcode a path to arm64-v8a - I assume that should actually be the path to whatever aot output we find?

Reply:
I think we could still leave this as APK right? Or is this code hit for iOS as well?

Reply:
this is used for every platform now, but most of the time it is just looking at a directory